### ISSUE TITLE
Embedding youtube video in product description is broken

### ISSUE THREAD

**Issue by hounw** [soflyy/oxygen-bugs-and-features#2530]
When you embed a youtube video using wordpress' native copy video url and paste in description box functionality the video iframe is loaded however oxygen somehow prevents wp from adding the needed divs/classes for it to work. It looks sort of ok on desktop but is completely broken in mobile.

https://oxygen-qwgxmazxn6mjl.oxygen-demo.qsandbox.me/product/test-prod/

1. copy youtube video url.
2. create a product
3. paste video in description
4. visualize on mobile

**Comment by KittenCodes** [comment]
@hounw The iframe has a fixed width and height, which is why it doesn't display correctly on mobile. To prevent that from occurring, you can add the following CSS to a stylesheet:

```css
iframe {
    max-width: 100%;
}
```

**Comment by hounw** [comment]
@KittenCodes thank you for your reply! While it is a good workaround upon further investigation I believe this should be a feature request, since most themes (even the most basic of them) handle this in a better manner. Do take into account that this copy-paste functionality is designed so that a content creator (not a web designer) can easily add videos in descriptions of products or other places.

You can check the sandbox again, I installed a popular theme and re-added the video. In this case it adds a wrapper to always make it full width and responsive. The default 2021 doesn't make it full width but has some code to make it responsive and preserve the aspect ratio.

In summary maybe it's not a bug but should definitely be a feature.

**Comment by KittenCodes** [comment]
@hounw I can see that there's some sort of JS on the Sandbox site that is setting the max-height of the iframe. It's being calculated based on the width of the viewport.

Would you like to open a new feature request for this?

One other workaround I can think of would be to add an ACF field so that your client can paste the video URL into it. You could then add a Video element to the design and retrieve that URL dynamically.

**Comment by hounw** [comment]
@KittenCodes YES! let's add a feature request of this... I would love to see oxygen work more on having standard stuff like not breaking other plugins, better defaults for woocommerce (like the image thumbnails under the product's main image) and stuff like this that makes it more usable for the average wordpress user.

I realize this might not be the focus of the company but you guys are seriously LIGHTYEARS away from any other builder for a web designer/developer, but if the project is dynamic and an end user will manage the site AFTER it has been designed, it's details like this that make it less than ideal...
